.ci/test_r_package.sh

```diff
@@ -112,7 +112,7 @@ cd ${BUILD_DIRECTORY}
 PKG_TARBALL="lightgbm_*.tar.gz"
 LOG_FILE_NAME="lightgbm.Rcheck/00check.log"
 if [[ $R_BUILD_TYPE == "cmake" ]]; then
-    Rscript build_r.R --skip-install || exit -1
+    Rscript build_r.R -j2 --skip-install || exit -1
```

Comment:
Why `j2`? We use `j4` in all our CI jobs, e.g.
https://github.com/microsoft/LightGBM/blob/aafedd8a1c57432ac3e38e4af236c00c6c54e84c/.ci/test.sh#L105
https://github.com/microsoft/LightGBM/blob/aafedd8a1c57432ac3e38e4af236c00c6c54e84c/.ci/test.sh#L216
Refer to #1745.

Reply:
Based on https://github.com/microsoft/LightGBM/pull/4525#issuecomment-904186660 and the investigation I did last night (https://github.com/microsoft/LightGBM/pull/4525#issuecomment-962540858), I found that using more threads than you have available CPUs doesn't provide much speedup compared to using exactly as many threads as CPUs, but does lead to higher memory consumption.

According to https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources, the VMs we get from GitHub Actions come with a single 2-core CPU. This is why I chose `-j2`.

All that said though, I think most of the benefit for CI comes from moving from single-threaded to multi-threaded, and the difference between `-j2` and  `-j4` shouldn't be large in that environment. And I don't think the memory consumption for compilation is going to be a problem, given your point about all other jobs in this project's CI using `-j4` already. I'll change these to `-j4` for consistency with the other CI jobs.